common/s3URLParts.go

```diff
@@ -114,6 +119,11 @@ func NewS3URLParts(u url.URL) (S3URLParts, error) {
 
 		if bucketEndIndex := strings.Index(path, "/"); bucketEndIndex != -1 {
 			up.BucketName = path[:bucketEndIndex]
+			if rawPath != "" {
+				up.RawBucketName = rawPath[:strings.Index(rawPath, "/")]
+			} else {
+				up.RawBucketName = up.BucketName
+			}
```

Comment:
I haven't understood yet why getting RawBucketName is necessary, and why getting it from rawPath like this is safe. (E.g. what if RawPath contains an escaped / ?  I haven't managed to think that through)

Can you explain a bit why its necessary and safe?

Reply:
If it were escaped in RawPath, it would be unescaped in path

https://play.golang.org/p/h5UGb923Qh4

Arguably using path is slightly less safe in a way?

Reply:
>if it were escaped in RawPath, it would be unescaped in path

How does that help, given that if RawPath exists, it becomes our only source of RawBucketName?

Reply:
I guess I don't really understand what we're trying to do here, in terms of why we need to obtain raw bucket name. It's just for the test in zc_enumerator right?

Reply:
THe test there would feel safer to me if, instead of being this:

```
if s3URLParts.RawBucketName == "" || strings.Contains(s3URLParts.RawBucketName, "*") {
```

it was this instead

```
if s3URLParts.BucketName == "" || strings.Contains(s3URLParts.BucketName, "*")  || strings.Contains(s3URLParts.RawBucketName, "*") {
```
I.e. use BucketName for the "is empty" part of the test, then check both Raw and non-raw for the "is star" part.  That feels safe to me with this current implementation of s3UrlParts.  What do you think?

Reply:
Raw bucket name wasn't actually necessary after the changes made above. This was more just mild internal panic.